packages/commerce-sdk-react/src/hooks/ShopperProducts/query.ts

```diff
@@ -7,61 +7,151 @@
 import {ApiClients, Argument, DataType} from '../types'
 import {useAsync} from '../useAsync'
 import useCommerceApi from '../useCommerceApi'
-import {UseQueryResult} from '@tanstack/react-query'
+import {UseQueryOptions, UseQueryResult} from '@tanstack/react-query'
 
 type Client = ApiClients['shopperProducts']
 
+type UseProductsParameters = NonNullable<Argument<Client['getProducts']>>['parameters']
+type UseProductsHeaders = NonNullable<Argument<Client['getProducts']>>['headers']
+type UseProductsArg = {headers?: UseProductsHeaders; rawResponse?: boolean} & UseProductsParameters
 /**
  * A hook for `ShopperProducts#getProducts`.
  * Allows access to multiple products by a single request. Only products that are online and assigned to a site catalog are returned. The maximum number of productIDs that can be requested are 24. Along with product details, the availability, images, price, promotions, and variations for the valid products will be included, as appropriate.
  * @see {@link https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-products?meta=getProducts} for more information about the API endpoint.
  * @see {@link https://salesforcecommercecloud.github.io/commerce-sdk-isomorphic/classes/shopperproducts.shopperproducts-1.html#getproducts} for more information on the parameters and returned data type.
  * @returns An object describing the state of the request.
  */
-export const useProducts = (
-    arg: Argument<Client['getProducts']>
-): UseQueryResult<DataType<Client['getProducts']>, Error> => {
+function useProducts(
+    arg: Omit<UseProductsArg, 'rawResponse'> & {rawResponse?: false},
+    options?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+): UseQueryResult<DataType<Client['getProducts']>, Error>
+function useProducts(
+    arg: Omit<UseProductsArg, 'rawResponse'> & {rawResponse: true},
+    options?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+): UseQueryResult<Response, Error>
+function useProducts(
+    arg: UseProductsArg,
+    options?: UseQueryOptions<DataType<Client['getProducts']> | Response, Error>
+) {
+    if (!arg.ids) {
+        throw new Error('ids is required for useProducts')
+    }
```

Comment:
what happens if you don't add this if statement?

Will commerce-sdk-isomorphic throw a similar error? Should we use that error? Because ultimately the schema is coming from the commerce-sdk-isomorphic `parameters`

Reply:
If we don't catch the error here,  it will hit the network and returns an error, since we are using `react-query`, by default it will retry 3 times before returning the error. I'd it is best if we can catch the error here before it hits the network. Here is what looks like if we don't catch that ids first
<img width="855" alt="image" src="https://user-images.githubusercontent.com/52219283/187315298-cb902cb4-c178-418f-9887-28412a6a67ce.png">